sed/dataset/dataset.py

```diff
@@ -0,0 +1,285 @@
+from pathlib import Path
+
+import requests
+
+from sed.core.config import load_config
+from sed.core.config import save_config
+from sed.core.logging import setup_logging
+from sed.core.user_dirs import construct_module_dirs
+
+# Configure logging
+logger = setup_logging(__name__)
+
+NAME = "datasets"
+user_paths = construct_module_dirs(NAME)
+json_path_user = user_paths["config"].joinpath(NAME).with_suffix(".json")
+json_path_module = Path(os.path.dirname(__file__)).joinpath(NAME).with_suffix(".json")
+
+
+def load_datasets_dict() -> dict:
+    """
+    Loads the datasets configuration dictionary from the user's datasets JSON file.
+
+    If the file does not exist, it copies the default datasets JSON file from the module
+    directory to the user's datasets directory.
+
+    Returns:
+        dict: The datasets dict loaded from user's datasets JSON file.
+    """
+    # check if datasets.json exists in user_config_dir
+    if not os.path.exists(json_path_user):
+        shutil.copy(json_path_module, json_path_user)
+    datasets = load_config(str(json_path_user))
+    return datasets
+
+
+def available_datasets() -> list:
+    """
+    Returns a list of available datasets.
+
+    Returns:
+        list: List of available datasets.
+    """
+    datasets = load_datasets_dict()
+    return list(datasets)
+
+
+def check_dataset_availability(data_name: str) -> dict:
+    """
+    Checks if the specified dataset is available in the predefined list of datasets.
+
+    Args:
+        data_name (str): The name of the dataset to check.
+
+    Returns:
+        dict: The dataset information if available.
+
+    Raises:
+        ValueError: If the dataset is not found in the predefined list.
+    """
+    datasets = load_datasets_dict()
+    if data_name not in datasets:
+        error_message = f"Data '{data_name}' is not available for fetching.\
+            Available datasets are: {available_datasets()}"
+        logger.error(error_message)
+        raise ValueError(error_message)
+    return datasets.get(data_name)
+
+
+def set_data_path(data_name: str, data_path: str, existing_data_path: str) -> str:
+    """
+    Determines and sets the data path for a dataset. If a data path is not provided,
+    it uses the existing data path or creates a new one. It also notifies the user
+    if the specified data path differs from an existing data path.
+
+    Args:
+        data_name (str): The name of the dataset.
+        data_path (str, optional): The desired path where the dataset should be stored.
+        existing_data_path (str, optional): The path where the dataset currently exists.
+
+    Returns:
+        str: The final data path for the dataset.
+    """
+    # Notify the user if data might already exist
+    if existing_data_path and data_path and existing_data_path != data_path:
+        logger.warning(
+            f'{data_name} data might already exists at "{existing_data_path}", '
+            "unless deleted manually.",
+        )
```

Comment:
Why is this a warning?

Reply:
Since log level for user is set to warning at default and I think this is important information to know. Otherwise you download the dataset again for no reason.

Reply:
Maybe I don't understand the logic well here. When is this warning supposed to be issued? How is existing_data_path determined by the caller, i.e. how do you find all possible existing versions of this data set?

Reply:
Currently, only one path is saved to existing version. When data is downloaded, extracted (and rearranged), that path is saved to the user config. The file list is also saved and can be compared to see if data is corrupted (maybe by user deleting something) or is how it is supposed to be.
I should extend this to include not just one path but multiple paths the data is downloaded to.